Cross-compiling for Windows on macOS fails

**Godot version:**
3.1, 3.2

**OS/device including version:**
```
$ sw_vers
ProductName:    Mac OS X
ProductVersion: 10.14.6
BuildVersion:   18G95
```
```
$ brew install mingw-w64
Warning: mingw-w64 6.0.0_2 is already installed and up-to-date
To reinstall 6.0.0_2, run `brew reinstall mingw-w64`
```

```
$ i686-w64-mingw32-gcc --version
i686-w64-mingw32-gcc (GCC) 9.1.0
(...)
```

**Issue description:**
I'm trying to cross-compilfe for Windows on macOS, referring to [Compiling for Windows].

I ran into `error: 'wcscpy_s' was not declared in this scope; did you mean 'wcscpy'?`

```
$ scons platform=windows tools=no target=release bits=64
(..omitted)
[Initial build] Compiling ==> platform/windows/os_windows.cpp
platform/windows/os_windows.cpp: In member function 'virtual Error OS_Windows::move_to_trash(const String&)':
platform/windows/os_windows.cpp:3213:2: error: 'wcscpy_s' was not declared in this scope; did you mean 'wcscpy'?
 3213 |  wcscpy_s(from, p_path.length() + 1, p_path.c_str());
      |  ^~~~~~~~
      |  wcscpy
scons: *** [platform/windows/os_windows.windows.opt.64.o] Error 1
scons: building terminated because of errors.
```

I investigated the error a little bit and found two ways to fix the issue ([reference](https://sourceforge.net/p/mingw-w64/mailman/mingw-w64-public/thread/06F7A3B7CD59E243A036828BAE5DF76F0267C4%40NA-MBX-01.mgc.mentorg.com/#messages)):
1) Specify `--enable-secure-api` on the header configuration step when compiling `mingw-w64`
2) Specify `-DMINGW_HAS_SECURE_API=1` on compilation.

Going for (1) can be a good way in terms of offloading the dependency management,
but (2) looks better for me because there seems to be no easy way to set `--endable-secure-api` when installing `mingw-w64` with `brew`, which means that we force developers to compile their own `mingw-w64` to cross compile, which is hard and time consuming

I tested and managed to fix the error by adding one line to `platform/windows/detect.py`
I'm preparing pull-requests of (2) for both `3.1` and `3.2`(master)

**Steps to reproduce:**
Just try to cross compile for Windows on macOS following [Compiling for Windows]

**Minimal reproduction project:**
Just clone the repository.


[Compiling for Windows]: https://docs.godotengine.org/en/3.1/development/compiling/compiling_for_windows.html#cross-compiling-for-windows-from-other-operating-systems

--- a/platform/windows/detect.py
+++ b/platform/windows/detect.py
@@ -350,7 +350,7 @@ def configure_mingw(env):
     env.Append(CPPDEFINES=[('WINVER', env['target_win_version']), ('_WIN32_WINNT', env['target_win_version'])])
     env.Append(LIBS=['mingw32', 'opengl32', 'dsound', 'ole32', 'd3d9', 'winmm', 'gdi32', 'iphlpapi', 'shlwapi', 'wsock32', 'ws2_32', 'kernel32', 'oleaut32', 'dinput8', 'dxguid', 'ksuser', 'imm32', 'bcrypt', 'avrt', 'uuid'])
 
-    env.Append(CPPDEFINES=['MINGW_ENABLED'])
+    env.Append(CPPDEFINES=['MINGW_ENABLED', ('MINGW_HAS_SECURE_API', 1)])
 
     # resrc
     env.Append(BUILDERS={'RES': env.Builder(action=build_res_file, suffix='.o', src_suffix='.rc')})

